### PR TITLE
Migrate ImmichLogger over to injected ILoggerRepository

### LINKED ISSUE #8824: Migrate ImmichLogger over to injected ILoggerRepository
#8823 introduces a new ILoggerRepository. All remaining usages of ImmichLogger should be migrated to use this new injected logger.

### PR DESCRIPTION
**Pull request by iamamansharma**
fixes  #8824

### CODE REVIEW

File: server/src/cores/storage.core.ts [hunk 1]
```diff
@@ -50,8 +50,10 @@ export class StorageCore {
     private cryptoRepository: ICryptoRepository,
     private repository: IStorageRepository,
     systemConfigRepository: ISystemConfigRepository,
+    @Inject(ILoggerRepository) private logger: ILoggerRepository,
```
Comment (danieldietzler) [inline server/src/cores/storage.core.ts:53]: You can't and don't want to have the inject there

File: server/src/cores/system-config.core.ts [hunk 2]
```diff
@@ -169,16 +169,20 @@ let instance: SystemConfigCore | null;
...
 export class SystemConfigCore {
-  private logger = new ImmichLogger(SystemConfigCore.name);
   private configCache: SystemConfigEntity<SystemConfigValue>[] | null = null;
 
   public config$ = new Subject<SystemConfig>();
 
-  private constructor(private repository: ISystemConfigRepository) {}
+  private constructor(
+    private repository: ISystemConfigRepository,
+    @Inject(ILoggerRepository) private logger: ILoggerRepository,
```
Comment (danieldietzler) [inline server/src/cores/system-config.core.ts:178]: Here the same thing. You don't want to get something injected when you call the constructor yourself (using the create method)

File: server/src/repositories/media.repository.ts [hunk 3]
```diff
@@ -21,8 +22,9 @@ sharp.cache({ files: 0 });
 
 @Instrumentation()
```
Comment (danieldietzler) [inline server/src/repositories/media.repository.ts:23]: You'll also need `@Injectable()` for it to work

File: server/src/repositories/metadata.repository.ts [hunk 4]
```diff
@@ -26,9 +26,10 @@ export class MetadataRepository implements IMetadataRepository {
     @Inject(ISystemMetadataRepository)
     private readonly systemMetadataRepository: ISystemMetadataRepository,
     @InjectDataSource() private dataSource: DataSource,
-  ) {}
-
-  private logger = new ImmichLogger(MetadataRepository.name);
+    @Inject(ILoggerRepository) private readonly logger: ILoggerRepository,
```
Comment (danieldietzler) [inline server/src/repositories/metadata.repository.ts:29]: Why do you give it the readonly attribute here?

File: server/src/repositories/search.repository.ts [hunk 5]
```diff
@@ -37,7 +36,9 @@ export class SearchRepository implements ISearchRepository {
     @InjectRepository(AssetFaceEntity) private assetFaceRepository: Repository<AssetFaceEntity>,
     @InjectRepository(SmartSearchEntity) private smartSearchRepository: Repository<SmartSearchEntity>,
     @InjectRepository(GeodataPlacesEntity) private readonly geodataPlacesRepository: Repository<GeodataPlacesEntity>,
+    @Inject(ILoggerRepository) private readonly logger: ILoggerRepository,
```
Comment (danieldietzler) [inline server/src/repositories/search.repository.ts:39]: Here as well?

File: server/src/repositories/storage.repository.ts [hunk 6]
```diff
@@ -13,13 +15,13 @@ import {
   WatchEvents,
 } from 'src/interfaces/storage.interface';
 import { Instrumentation } from 'src/utils/instrumentation';
-import { ImmichLogger } from 'src/utils/logger';
 import { mimeTypes } from 'src/utils/mime-types';
 
 @Instrumentation()
```
Comment (danieldietzler) [inline server/src/repositories/storage.repository.ts:20]: `@Injectable()`

**Review by danieldietzler — COMMENTED** [review]
I wonder if we should call the logger variables `loggerRepository` for consistency or no...? @jrasm91

File: server/src/cores/storage.core.ts (hunk 1 above)
Comment (iamamansharma) [reply server/src/cores/storage.core.ts:53]: it seems to me that we wanna inject the new logger .
ref: https://github.com/immich-app/immich/pull/8823

**Comment by jrasm91** [comment]
> I wonder if we should call the logger variables `loggerRepository` for consistency or no...? @jrasm91

I don't have a strong preference, but that might make it more consistent, so I'm not against doing it in a follow-up PR.

File: server/src/repositories/logger.repository.ts [hunk 7]
```diff
@@ -1,10 +1,10 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Scope } from '@nestjs/common';
 import { ClsService } from 'nestjs-cls';
 import { LogLevel } from 'src/entities/system-config.entity';
 import { ILoggerRepository } from 'src/interfaces/logger.interface';
 import { ImmichLogger } from 'src/utils/logger';
 
-@Injectable()
+@Injectable({ scope: Scope.TRANSIENT })
```
Comment (jrasm91) [inline server/src/repositories/logger.repository.ts:7]: This is necessary so each injected instance is independent and `.setContext` in one service doesn't override that of another

File: server/src/services/database.service.spec.ts [hunk 8]
```diff
@@ -23,30 +25,18 @@ describe(DatabaseService.name, () => {
...
     [{ vectorExt: DatabaseExtension.VECTOR, extName: 'pgvector', minVersion: new Version(0, 5, 0) }],
   ] as const)('init', ({ vectorExt, extName, minVersion }) => {
-    let fatalLog: MockInstance;
-    let errorLog: MockInstance;
-    let warnLog: MockInstance;
-
     beforeEach(() => {
-      fatalLog = vitest.spyOn(ImmichLogger.prototype, 'fatal');
-      errorLog = vitest.spyOn(ImmichLogger.prototype, 'error');
-      warnLog = vitest.spyOn(ImmichLogger.prototype, 'warn');
```
Comment (jrasm91) [inline server/src/services/database.service.spec.ts:33]: these tests are so much nicer now that the logger in dependency injected instead of using a spy

**Review by jrasm91 — APPROVED** [review]
Ty!

**Review by danieldietzler — APPROVED** [review]
LGTM!

**Pull request merged**